File: app/src/main/java/com/absinthe/anywhere_/view/AnywhereDialogFragment.java
package com.absinthe.anywhere_.view;

import android.content.DialogInterface;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.FragmentManager;

import com.absinthe.anywhere_.utils.manager.DialogStack;

public class AnywhereDialogFragment extends DialogFragment {

    private boolean isDismissParent = false;
    private OnDismissListener mListener;

    @Override
    public void show(@NonNull FragmentManager manager, @Nullable String tag) {
        super.show(manager, tag);
        DialogStack.push(this);
    }

    @Override
    public void onDismiss(@NonNull DialogInterface dialog) {
        super.onDismiss(dialog);
        DialogStack.pop();
        if (isDismissParent) {
            DialogStack.pop();
        }
        if (mListener != null) {
            mListener.onDismiss();
        }
    }

    protected void setWrapOnDismissListener(OnDismissListener listener) {
        mListener = listener;
    }

    protected void setDismissParent(boolean flag) {
        isDismissParent = flag;
    }

    public interface OnDismissListener {
        void onDismiss();
    }
}

File: app/src/main/java/com/absinthe/anywhere_/view/app/AnywhereDialogFragment.kt
package com.absinthe.anywhere_.view.app

import android.content.DialogInterface
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.FragmentManager
import com.absinthe.anywhere_.utils.manager.DialogStack
import timber.log.Timber

open class AnywhereDialogFragment : DialogFragment() {

    private var isDismissParent = false
    private var mListener: OnDismissListener? = null

    override fun show(manager: FragmentManager, tag: String?) {
        try {
            super.show(manager, tag)
        } catch (e: IllegalStateException) {
            Timber.e(e)
        }
        DialogStack.push(this)
    }

    override fun onDismiss(dialog: DialogInterface) {
        super.onDismiss(dialog)

        DialogStack.pop()
        if (isDismissParent) {
            DialogStack.pop()
        }
        mListener?.onDismiss()

    }

    protected fun setWrapOnDismissListener(listener: OnDismissListener?) {
        mListener = listener
    }

    protected fun setDismissParent(flag: Boolean) {
        isDismissParent = flag
    }

    interface OnDismissListener {
        fun onDismiss()
    }
}